src/plugins/intel_cpu/src/utils/plain_tensor.hpp

```diff
@@ -415,25 +415,13 @@ struct PlainTensor {
     [[nodiscard]] int64_t offset(I i, Is... indices) const {
         return i * m_strides[dim] + offset<dim + 1>(indices...);
     }
-    template <typename DT, typename... Is>
-    [[nodiscard]] DT* ptr(Is... indices) const {
-        return reinterpret_cast<DT*>(m_ptr.get()) + offset<0>(indices...);
-    }
 
-    template <typename DT,
-              ov::element::Type_t SRC_PREC,
-              std::enable_if_t<SRC_PREC != ov::element::u4, bool> = true,
-              typename... Is>
-    DT* ptr(Is... indices) const {
-        return reinterpret_cast<DT*>(m_ptr.get()) + offset<0>(indices...);
-    }
+    template <typename DT, ov::element::Type_t SRC_PREC = ov::element::u8, typename... Is>
+    [[nodiscard]] DT* ptr(Is... indices) const {
+        static constexpr size_t stride_div = SRC_PREC == ov::element::u4 ? 2 : 1;
```

Comment:
The static for single constexpr value can be removed it should not change anything

can the whole function be constexpr?

Reply:
Theoretically can (requires more updates), but this is not the goal of the PR.